index.js

```diff
@@ -180,2669 +180,2588 @@
-                   (index_ => f => t => propPath => s =>
-                      index_ === index ? f (s) : ' '.repeat (s.length))
-      }\nExpected ${
-        numArgs (numArgsExpected)
-      } but received ${
-        numArgs (args.length)
-      }${
-        /* istanbul ignore next */
-        args.length === 0 ? '.\n' : ':\n\n' + Z.foldMap (
-          globalThis.String,
-          x => `  - ${show (x)}\n`,
-          args
-        )
-      }`
-    )
-  );
+//# Undefined :: Type
+//.
+//. Type whose sole member is `undefined`.
+export const Undefined = NullaryTypeWithUrl
+  ('Undefined')
+  ([])
+  (x => type (x) === 'Undefined');
 
-  //  constraintsRepr :: ... -> String
-  const constraintsRepr = (
-    constraints,    // :: StrMap (Array TypeClass)
-    outer,          // :: String -> String
-    inner           // :: String -> TypeClass -> String -> String
-  ) => {
-    const reprs = Z.chain (
-      k => (
-        constraints[k].map (typeClass =>
-          inner (k) (typeClass) (`${stripNamespace (typeClass)} ${k}`)
-        )
+//# env :: Array Type
+//.
+//. An array of [types][]:
+//.
+//.   - <code>[AnyFunction](#AnyFunction)</code>
+//.   - <code>[Arguments](#Arguments)</code>
+//.   - <code>[Array](#Array) ([Unknown][])</code>
+//.   - <code>[Array2](#Array2) ([Unknown][]) ([Unknown][])</code>
+//.   - <code>[Boolean](#Boolean)</code>
+//.   - <code>[Buffer](#Buffer)</code>
+//.   - <code>[Date](#Date)</code>
+//.   - <code>[Descending](#Descending) ([Unknown][])</code>
+//.   - <code>[Either](#Either) ([Unknown][]) ([Unknown][])</code>
+//.   - <code>[Error](#Error)</code>
+//.   - <code>[Fn](#Fn) ([Unknown][]) ([Unknown][])</code>
+//.   - <code>[HtmlElement](#HtmlElement)</code>
+//.   - <code>[Identity](#Identity) ([Unknown][])</code>
+//.   - <code>[JsMap](#JsMap) ([Unknown][]) ([Unknown][])</code>
+//.   - <code>[JsSet](#JsSet) ([Unknown][])</code>
+//.   - <code>[Maybe](#Maybe) ([Unknown][])</code>
+//.   - <code>[Module](#Module)</code>
+//.   - <code>[Null](#Null)</code>
+//.   - <code>[Number](#Number)</code>
+//.   - <code>[Object](#Object)</code>
+//.   - <code>[Pair](#Pair) ([Unknown][]) ([Unknown][])</code>
+//.   - <code>[RegExp](#RegExp)</code>
+//.   - <code>[StrMap](#StrMap) ([Unknown][])</code>
+//.   - <code>[String](#String)</code>
+//.   - <code>[Symbol](#Symbol)</code>
+//.   - <code>[Type](#Type)</code>
+//.   - <code>[TypeClass](#TypeClass)</code>
+//.   - <code>[Undefined](#Undefined)</code>
+env.push (
+  AnyFunction,
+  Arguments,
+  Array (Unknown),
+  Array2 (Unknown) (Unknown),
+  Boolean,
+  Buffer,
+  Date,
+  Descending (Unknown),
+  Either (Unknown) (Unknown),
+  Error,
+  Fn (Unknown) (Unknown),
+  HtmlElement,
+  Identity (Unknown),
+  JsMap (Unknown) (Unknown),
+  JsSet (Unknown),
+  Maybe (Unknown),
+  Module,
+  Null,
+  Number,
+  Object,
+  Pair (Unknown) (Unknown),
+  RegExp,
+  StrMap (Unknown),
+  String,
+  Symbol,
+  Type,
+  TypeClass,
+  Undefined,
+);
+export {env};
```

Comment:
Why not have the `export` at the `const` statement?

Reply:
I tried to move as few things as possible from their current—presumably intended—locations. I moved `env` because I could see no other option.

Whether we export `env` here or there is not important to me. Do you have a preference?

Reply:
My preference would be to have the export of `env` by its definition, irrespective of whether it's being mutated later.